environment/lib/eth-proof-extractor/index.js

```diff
@@ -57,42 +57,36 @@ class EthProofExtractor {
         return tree;
     }
 
-    async extractProof (block, tree, transactionIndex) {
+    async extractProof (web3, block, tree, transactionIndex) {
         const [, , stack] = await promisfy(tree.findPath, tree)(encode(transactionIndex));
+
+        const blockData = await web3.eth.getBlock(block.number);
+        // Correctly compose and encode the header.
+        const header = [
+            blockData.parentHash,
+            blockData.sha3Uncles,
+            blockData.miner,
+            blockData.stateRoot,
+            blockData.transactionsRoot,
+            blockData.receiptsRoot,
+            blockData.logsBloom,
+            blockData.difficulty == "0" ? "0x": web3.utils.toHex(blockData.difficulty),
```

Comment:
@nearmax I did some further investigation, this is correct, but in general, all other places that use number that could be 0, such as gasUsed, also have this problem, so I think we should add an additional pass to eth-object/eth-util-lite (will fork it), so it => [strings].map(toHex).map(if "0x0" else return =>"0x") => rlp encode will be correct

Reply:
Hmm looks i'm wrong, gasUsed is number, should be '0x0' when it's 0. But difficulty is a string, should be '0x' if it's '0' 🤔